### PR TITLE
os/bluestore: support for FreeBSD

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by wjwithagen** [comment]
@tchaikov 
It is running in Jenkins atm.:
    http://cephdev.digiware.nl:8180/jenkins/job/ceph-master/2888/console

File: CMakeLists.txt [hunk 1]
```diff
@@ -285,7 +290,7 @@ if(WITH_PMEM)
 endif()
 
 if(WITH_BLUESTORE)
-  if(NOT AIO_FOUND AND NOT WITH_SPDK AND NOT WITH_PMEM)
+  if(NOT AIO_FOUND AND NOT HAVE_POSIXAIO AND NOT WITH_SPDK AND NOT WITH_PMEM)
```
Comment (wjwithagen) [inline CMakeLists.txt]: @tchaikov 
I think this needs to change into:
```
 if(NOT (AIO_FOUND OR HAVE_POSIXAIO) AND NOT WITH_SPDK AND NOT WITH_PMEM)
```
So the ORed term is true if any of the 2 AIOs is available.

Comment (tchaikov) [reply CMakeLists.txt]: no, i think it's correct and it follows the existing pattern.

**Comment by tchaikov** [comment]
@wjwithagen fixed and repushed.

File: CMakeLists.txt (hunk 1 above)
Comment (wjwithagen) [reply CMakeLists.txt]: @tchaikov 
Running it again.

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-12-20_12:03:44-rados-wip-kefu2-testing-2018-12-20-1729-distro-basic-smithi/

**Comment by liewegas** [comment]
s/ceph-aio/ceph_aio/... otherwise looks good!

**Comment by tchaikov** [comment]
we need to rename `ceph-aio.h` to `ceph_aio.h`.

**Comment by tchaikov** [comment]
```
Start 151: unittest_rgw_reshard_wait
140/161 Test #151: unittest_rgw_reshard_wait ...............***Failed    0.50 sec
Running main() from gmock_main.cc
[==========] Running 5 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 5 tests from ReshardWait
[ RUN      ] ReshardWait.wait_block
[       OK ] ReshardWait.wait_block (10 ms)
[ RUN      ] ReshardWait.stop_block
[       OK ] ReshardWait.stop_block (10 ms)
[ RUN      ] ReshardWait.wait_yield
/home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/test/rgw/test_rgw_reshard_wait.cc:69: Failure
      Expected: 1u
      Which is: 1
To be equal to: context.poll()
      Which is: 2
/home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/test/rgw/test_rgw_reshard_wait.cc:70: Failure
Value of: context.stopped()
  Actual: true
Expected: false
/home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/test/rgw/test_rgw_reshard_wait.cc:73: Failure
      Expected: 1u
      Which is: 1
To be equal to: context.run_one()
      Which is: 0
/home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/test/rgw/test_rgw_reshard_wait.cc:77: Failure
Expected: (wait_duration) <= (elapsed), actual: 0.01s vs 0.000180293s
[  FAILED  ] ReshardWait.wait_yield (1 ms)
```

**Comment by tchaikov** [comment]
@cbodley does this look like a genuine issue?

**Comment by cbodley** [comment]
hi @tchaikov, it looks like `HAVE_BOOST_CONTEXT` isn't defined, so the call to `RGWReshardWait::wait()` is sleeping on a condition variable instead of yielding the coroutine here

was this build configured by `do_freebsd.sh`? i don't see anything in there that disables the `WITH_BOOST_CONTEXT` option

i've opened https://github.com/ceph/ceph/pull/25744 to disable these coroutine tests when `HAVE_BOOST_CONTEXT` isn't defined

**Comment by tchaikov** [comment]
@cbodley no, it was on our arm64 builder. and that was the first time i spotted this failure. will let you known if it surfaces again.

> i don't see anything in there that disables the `WITH_BOOST_CONTEXT` option

neither do i. so it must have been something else..

**Pull request merged**